test/helpers/position_assertions.cc

```diff
@@ -1129,4 +1130,119 @@ void ApplyCodeActionAssertion::check(const UnorderedMap<std::string, std::shared
     }
 }
 
+SymbolSearchAssertion::SymbolSearchAssertion(string_view filename, unique_ptr<Range> &range, int assertionLine,
+                                             string_view query)
+    : RangeAssertion(filename, range, assertionLine), query(query) {}
+
+shared_ptr<SymbolSearchAssertion> SymbolSearchAssertion::make(string_view filename, unique_ptr<Range> &range,
+                                                              int assertionLine, string_view assertionContents,
+                                                              string_view assertionType) {
+    static const regex contentsRegex(R"(^\s*\"([^\"]+)\"\s*$)");
+
+    smatch matches;
+    string assertionContentsString = string(assertionContents);
+    if (regex_search(assertionContentsString, matches, contentsRegex)) {
+        auto query = matches[1].str();
+        return make_shared<SymbolSearchAssertion>(filename, range, assertionLine, query);
+    }
+
+    ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1) << fmt::format(
+        "Improperly formatted assertion. Expected 'symbol-search: \"<query>\"'. Found '{}'", assertionContents);
+
+    return nullptr;
+}
+
+void SymbolSearchAssertion::checkAllForQuery(std::string query,
+                                             const vector<shared_ptr<SymbolSearchAssertion>> &assertions,
+                                             const UnorderedMap<string, shared_ptr<core::File>> &sourceFileContents,
+                                             LSPWrapper &lspWrapper, int &nextId, string_view uriPrefix,
+                                             string errorPrefix) {
+    const int id = nextId++;
+    const int lineWidth = 6; // ensure sort order == numerical order (if file is less than a million lines)
+
+    std::map<std::string, vector<shared_ptr<SymbolSearchAssertion>>> assertionsByLine;
+    for (auto assertion : assertions) {
+        auto location = assertion->getLocation(uriPrefix);
+        auto line = fmt::format("{}:{:{}}", location->uri, location->range->start->line, lineWidth);
+        assertionsByLine[line].push_back(assertion);
+    }
+
+    // Request results from LSP
+    auto responses = lspWrapper.getLSPResponsesFor(makeWorkspaceSymbolRequest(id, query));
+    ASSERT_EQ(1, responses.size()) << "Unexpected number of responses to a `textDocument/definition` request.";
```

Comment:
```suggestion
    ASSERT_EQ(1, responses.size()) << "Unexpected number of responses to a `workspace/symbol` request.";
```

Reply:
Oh, duh!  Will fix.  Although this does make me wonder: why does the LSP support multiple responses in the first place?  Is it a mechanism by which the LSP can return incremental progress to the client to reduce latency before the first results start showing up?  If so, would it make sense to ditch this assertion and accept/aggregate multiple responses?